src/flag_gems/ops/sigmoid.py

```diff
@@ -27,9 +27,13 @@ class Sigmoid(torch.autograd.Function):
     @staticmethod
     def forward(ctx, A):
         logging.debug("GEMS SIGMOID FORWARD")
-        out = sigmoid_forward(A.to(torch.float32))
-        ctx.save_for_backward(out)
-        return out.to(A.dtype)
+        if A.requires_grad is True:
+            out = sigmoid_forward(A.to(torch.float32))
+            ctx.save_for_backward(out)
+            return out.to(A.dtype)
+        else:
+            out = sigmoid_forward(A)
+            return out
```

Comment:
this may render the results in different precision given grad and no grad context. Shall we just keep the old version?

Reply:
input in any data types is upcast to tl.float32 in triton kernel, thus the precision of computing is the same. and for half precision, even if we upcast the input tensor to torch.float32 before calling kernel, it will be downcast when returning.